### DISCUSSION TITLE
UX of the songsheet view

### DISCUSSION THREAD

**Discussion by edwinzap** [chordbook/chordbook#1352] [General]
I create this discussion to chat together about how to make the best UX for the songsheet view.
Here are some ideas to start:

# Introduction
As a musician, if I open the songsheet view, it is only to **play** the song. That means that:
- Important stuff like title, artist,... should be seen before opening this view. Once I start playing, I don't really need these informations anymore.
- This view should contains only necessary informations to **play** the song => minimilast. Fullscreen mode would be nice to 
- The view should be mobile first: I will often play on small devices like phones or tablets but not often on a computer.
- The view should be customizable to my needs: chords and text size, colors,... zoom-in/out, show/hide tabs,...
- The view should be customizable depending on my experience. A beginner needs chord charts, the ability to simplify the chords, to be guided to have the best experience as possible. An experienced musician doesn't need this extra features that just takes unnecessary place on the screen.
- Some features need to be always available while playing (autoscroll, zoom-in/out, play/pause youtube music, chord charts) but other features are only set before playing (transpose, view customization). The "always available" features should be very easy to reach. This means that, on mobile (but also on desktop), the controls should be **right under the fingers** (so preferably at the bottom right/left (if the musician is righ/left sided).
- The preferences and customizations should be **user** or song **specific** and be saved. For example, if I decide to not show the chord charts, it should then be the case everywhere (user specific). But if I change the default speed of the autoscroll, it should be saved (song specific). Choosing between song/user specific could be discussed. We could also provide the ability to **"Set as default"** (like the text sizes and colors).

# Autoscroll
The autoscroll feature could have a speed based on the song duration (meta). If no duration is specify, the duration could be either the average one (214 seconds) or a duration based on the song content (maybe with AI).

Playing along with the youtube music would be nice. The duration would then be based on the youtube song duration. Starting/stopping the autoscroll would also start/stop the music.

The musician should be able to easily:
- start/stop autoscroll
- adapt the speed

# Transpose
Transposing is used in two cases:
- I cannot sing so high/low so I change the key
- As a guitar (or piano if I have this feature on my keyboard) player, I'm using a capo to have easier chords. 

After transposing, there are always:
- The original key in which I should be able to go back easily (reset)
- The new key (I **hear**, the "real" key)
- The new key (I **see**, the chords key)

For example:
I'm playing a song with an original key of **C major** with these chords: **C Am G**.
This is to high for me so I will tranpose it 2 semi tones lower. This gives me a new key: A# or Bb major (I will use Bb for the example)
The new chords are: **Bb Gm F**.
These are difficult chords on the guitar so I want to use a capo. One possibility is to use a capo 3. This gives me then the chords: **G Em D**.

It would be really nice to have a simple tool that help the musician choosing the right key. We can maybe be inspired by this website: https://www.mychordbook.com/chords/ed-sheeran/perfect

![image](https://github.com/chordbook/chordbook/assets/7621249/a03acfde-851a-4313-b2d2-7ef053b6ca3d)

For keys with # or b, there are always 2 possible keys: # or b. For example, the key Eb major can also be D# major. A key is either major or minor.

# Chord charts
Chord charts is a nice feature for beginners. There are 3 main instruments that are using chords:
- guitar
- ukulele
- piano
The user should be able to select the default instrument and eventually remove some instruments. For example, I only play piano and guitar so in my Settings, I would disable the ukulele from my instrument list.

There is a lot of possibilities (positions) for the same chord. The musician should be able to see these other possibilities.

Chord charts are used to help the musician while playing. That's why these chord charts should always be visible. One advanced feature would be to show only charts for the chords that are currently visible on the screen (I don't need to see chords that I don't need to play right now).

The chord charts could also be visible on mouse click (or hover) on the chord.

Allow experienced musicians to hide this beginner feature.

Finally, it would (maybe) be nice to allow the user to resize the chord charts panel to have more place on the screen for the song lyrics.

**Comment by edwinzap** [comment]
As a piano player, songs with capo is always a problem. Would be nice to have a feature to directly remove the capo and change the key accordingly. Could also be an option in the preferences to always show the song in the original key (so without capo).

**Comment by bkeepers** [comment]
This is all really great! I don't disagree with anything. Some things I'll have more thoughts as we're implementing the specifics, but your general philosophy is in line with the direction I am working toward.
